### PR TITLE
Mute clippy warnings in a few places, apply the suggestions in remaining cases.

### PR DESCRIPTION
**Pull request by zbraniecki**
I applied the clippy suggestion to change weekday names, and muted it in cases where we want to keep `UTS35` or `ICU4X` etc.

Also, nightly clippy warned about bare trait objects being deprecated so added `dyn`.

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

File: components/capi/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // This file is part of ICU4X. For terms of use, please see the file
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+#![allow(clippy::upper_case_acronyms)]
```
Comment (sffc) [inline components/capi/src/lib.rs:4]: You need an empty line after the copyright header

**Review by sffc — DISMISSED** [review]
Thanks for the fixes.  This is good as-is, but did you investigate why the CI wasn't failing with a hard error status?

**Comment by zbraniecki** [comment]
> Thanks for the fixes. This is good as-is, but did you investigate why the CI wasn't failing with a hard error status?

nope. Didn't have time for that

File: components/capi/src/lib.rs (hunk 1 above)
Comment (zbraniecki) [reply components/capi/src/lib.rs:5]: fixed!

File: use_pattern.diff [hunk 2]
```diff
@@ -0,0 +1,87 @@
+diff --git a/Cargo.lock b/Cargo.lock
```
Comment (sffc) [inline use_pattern.diff:1]: Delete this extra file please

File: lifetime_provider.diff [hunk 3]
```diff
@@ -0,0 +1,430 @@
+diff --git a/components/datetime/src/lib.rs b/components/datetime/src/lib.rs
```
Comment (sffc) [inline lifetime_provider.diff:1]: Delete this extra file please

**Review by sffc — CHANGES_REQUESTED** [review]

File: use_pattern.diff (hunk 2 above)
Comment (zbraniecki) [reply use_pattern.diff]: fixed!

File: lifetime_provider.diff (hunk 3 above)
Comment (zbraniecki) [reply lifetime_provider.diff]: fixed!

**Review by sffc — APPROVED** [review]

**Pull request merged**